### ISSUE TITLE
Add registers

### ISSUE THREAD

**Issue by dyerti** [kyoheiu/felix#204]
Is there any interest for support registers? I'd be open to contributing if there is interest.

My use case would be disjoint file manipulation and it seems like it would be a natural extension of the existing support.

Existing yank and put would modify the `""` register. Named registers (`"a`) could be used to collect file selections when coupled with register appending (`"A`).

```
:reg a, " - list the contents of register a and the default register
"ay (select mode) - yank selected files into register a
"ad (select mode) - delete selected files and yank into register a
"Ay (select mode) - yank selected files and append into register a
"Ad (select mode) - delete selected files, yank and append into register a
"ayy - yank selected file into register a
"add - delete and yank select file into register a
"Ayy - yank selected file and append into register a
"Ayy - delete selected file, yank and append into register a
"ap - put contents of register a into current directory
```

**Comment by kyoheiu** [comment]
Sorry for being late.
That'd be interesting and useful!

What we have to think would be:

- Which type of register should be covered? Numbered only, or as you suggest, numbered and named? Numbered register is simpler and intuitive, and may be easy to implement IMO, on the other hand named register offers more flexible way to manipulate files.

- How should we think about the data structure? Now `registered` field is `Vec<ItemInfo>`, but with the register feature that may be `BTreeMap<Usize, Vec<ItemInfo>>` or `BTreeMap<Usize, Vec<ItemInfo>>`: This would vary based on the register type.

**Comment by dyerti** [comment]
I was thinking numbered and named, so using something like `BTreeMap<String, Vec<ItemInfo>>`.

I could start with numbered for the PoC and still use `BTreeMap<String, Vec<ItemInfo>>`. It would make the changes needed in event loop smaller.

**Comment by kyoheiu** [comment]
Yeah, that's good if named is planned.
(`vec_deque` would be enough if numbered only though)

I'm ready to implement it, but what's your thought? Can I assign myself to this?

**Comment by dyerti** [comment]
Sounds good. Go for it!

**Comment by kyoheiu** [comment]
Implemented by #215.
It needed much refactoring but it feels now way, way better than before! Thank you.
If you're interested, please build from `feature-reg` branch and tinker with it.
After debugging I'll release new version.

**Comment by kyoheiu** [comment]
Merged & released as v2.4.0.
